src/node.rs

```diff
@@ -338,8 +422,36 @@ async fn socks5_once(
                         ))
                         .await?;
                 }
-                Socks5Fallback::SimpleProxy { remote } => {
-                    let remote_stream = pool.connect(remote, addr.as_bytes()).await?;
+                Socks5Fallback::SimpleProxy { exit_nodes } => {
+                    // let random_port = rand::thread_rng().gen_range(1024..65535);
```

Comment:
There's probably a smarter way to handle port selection. I imagine we want to request a range of (common) ports. This should work nicely with the port-to-fingerprint mapping in the `ExitRegistry`, since it's a `BTreeMap`